packages/oui-header-tabs/src/header-tabs.controller.js

```diff
@@ -0,0 +1,59 @@
+import _ from "lodash";
+export default class {
+    constructor ($state, $transitions) {
+        "ngInject";
+        this.$state = $state;
+        this.onTransitionHandler = $transitions.onSuccess({}, () => {
+            this.updateActiveTab();
+        });
+    }
+
+    $onInit () {
+        this.tabs = [];
+        this.activeTab = null;
+    }
+
+    updateActiveTab () {
+        const currentTab = _.find(this.tabs, tab => tab.state && this.$state.includes(tab.state));
+        if (currentTab && currentTab !== this.activeTab) {
```

Comment:
Isn't it the same as `currentTab !== this.activeTab` ?

If `currentTab === undefined` then we go to the `else if` either way (`this.activeTab` can't be `undefined`)

Also, you are comparing objects, it feels dangerous to me but I don't know, I might be paranoid but I would have methods to:
* determine if a tab is currently active
* determine if two tabs are equal

Reply:
both of these can be undefined, in that case selectTab must not be called